firestore/query_test.go

```diff
@@ -448,8 +459,41 @@ func TestQueryToProto(t *testing.T) {
 	}
 }
 
+// Convert a Query to a Proto and back again verifying roundtripping
+func TestQueryFromProtoRoundTrip(t *testing.T) {
+	c := &Client{projectID: "P", databaseID: "DB"}
+
+	for _, test := range createTestScenarios(t) {
+		fmt.Println(test.desc)
+		proto, err := test.in.ToProto()
+		if err != nil {
+			t.Fatalf("%s: %v", test.desc, err)
+			continue
+		}
+		fmt.Printf("proto: %v\n", proto)
+		got, err := c.Query().FromProto(proto)
+		if err != nil {
+			t.Fatalf("%s: %v", test.desc, err)
+			continue
+		}
+
+		want := test.in
+		gotProto, err := got.ToProto()
+		fmt.Println(gotProto)
+		if err != nil {
+			t.Fatalf("%s: %v", test.desc, err)
+		}
+
+		// Compare protos before and after taking to a query. proto -> query -> proto.
+		if diff := cmp.Diff(gotProto, proto, protocmp.Transform()); diff != "" {
```

Comment:
@schmidt-sebastian resolving your other comment, but this is where we primarily are testing the serialization. We had a dataset testing to proto. I added a case that goes query -> proto -> query -> proto and compares both protos to verify they are the same.

Reply:
I saw that too late and forgot to close my previous comment :)